awscli/clidriver.py

```diff
@@ -379,15 +366,17 @@ def __call__(self, args, parsed_globals):
 
     def _create_command_table(self):
         command_table = OrderedDict()
-        service_object = self._get_service_object()
-        for operation_object in service_object.operations:
-            cli_name = xform_name(operation_object.name, '-')
+        service_model = self._get_service_model()
```

Comment:
Yeah, that looks to be true. We *need* the service_name (like the one you use to create a client) in order to get the waiter model. We cannot rely on the ``name`` property. I cannot think of any way around it. Extracting the service name from the event name is unreliable as well.